android/libraries/rib-base/src/main/java/com/badoo/ribs/core/routing/configuration/ConfigurationCommandCreator.kt

```diff
@@ -0,0 +1,205 @@
+                commands += backStackElement.removeAllOverlays(contentKey).reversed()
+                if (realIndex != lastIndex) {
+                    commands += backStackElement.deactivateAllOverlays(contentKey).reversed()
+                }
+
+                commands
+            }
+            .flatten()
+            .reversed()
+
+    }
+
+    private fun <C : Parcelable> List<BackStackElement<C>>.addFrom(targetIdxExclusive: Int): List<ConfigurationCommand<C>> {
+        val offset = targetIdxExclusive + 1
+        return subList(offset, size)
+            .mapIndexed { contentIndex, backStackElement ->
+                val commands = mutableListOf<ConfigurationCommand<C>>()
+                val contentKey = Content(offset + contentIndex)
+                commands += Add(contentKey, backStackElement.configuration)
+                commands += backStackElement.addAllOverlays(contentKey)
+                commands
+            }
+            .flatten()
+    }
+
+    private fun <C : Parcelable> List<BackStackElement<C>>.activateIfNeeded(other: List<BackStackElement<C>>): List<ConfigurationCommand<C>> =
+        when {
+            isNotEmpty() && contentListHasChanged(other) -> {
+                val commands = mutableListOf<ConfigurationCommand<C>>()
+                val contentKey = Content(lastIndex)
+                commands += Activate(contentKey)
+                commands += last().activateAllOverlays(contentKey)
+                commands
+            }
+            else -> emptyList()
+        }
+
+    private fun <C : Parcelable> BackStackElement<C>.addAllOverlays(content: Content): List<ConfigurationCommand<C>> =
+        overlays
+            .mapIndexed { overlayIndex, configuration ->
+                Add(Overlay(Key(content, overlayIndex)), configuration)
+            }
+
+    private fun <C : Parcelable> BackStackElement<C>.activateAllOverlays(contentKey: Content): List<ConfigurationCommand<C>> =
+        overlays
+            .mapIndexed { overlayIndex, _ ->
+                Activate<C>(Overlay(Key(contentKey, overlayIndex)))
+            }
+
+    private fun <C : Parcelable> BackStackElement<C>.deactivateAllOverlays(contentKey: Content): List<ConfigurationCommand<C>> =
+        overlays
+            .mapIndexed { overlayIndex, _ ->
+                Deactivate<C>(Overlay(Key(contentKey, overlayIndex)))
+            }
+            .reversed()
+
+    private fun <C : Parcelable> BackStackElement<C>.removeAllOverlays(contentKey: Content): List<ConfigurationCommand<C>> =
+        overlays
+            .mapIndexed { overlayIndex, _ ->
+                Remove<C>(Overlay(Key(contentKey, overlayIndex)))
+            }
+            .reversed()
+
+    private fun <C : Parcelable> List<BackStackElement<C>>.overlayDiff(other: List<BackStackElement<C>>): List<ConfigurationCommand<C>> {
+        val commands = mutableListOf<ConfigurationCommand<C>>()
+
+        forEachIndexed { index, myElement ->
```

Comment:
`myElement` -> `element`?

Reply:
Since there's `otherElement` in the same scope, makes it immediately understandable for me to keep the current naming.